src/jobQueue.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  run,
+  quickAddJob,
+} from 'graphile-worker';
+import { processBulkUpload } from './tasks';
+import type { ProcessBulkUploadJobPayload } from './types';
+
+enum JobType {
+  PROCESS_BULK_UPLOAD = 'processBulkUpload',
+}
+
+export const startJobQueue = async () => {
+  const runner = await run({
+    concurrency: 5,
+    noHandleSignals: false,
+    pollInterval: 1000,
+    taskList: {
+      processBulkUpload,
+    },
+  });
+  await runner.promise;
+};
+
+const addJob = async (
+  jobType: JobType,
+  payload: unknown,
+) => {
+  await quickAddJob(
+    {},
+    jobType,
+    payload,
+  );
+};
```

Comment:
Now that we're not specifying any connection parameters, do we still need to use `quickAddJob` instead of `addJob`? From the [documentation on adding jobs](https://worker.graphile.org/docs/library/add-job),

> `quickAddJob` is similar to `addJob`, but accepts an additional initial parameter describing how to connect to the database.

It's nice to type-check that the job type is known, but I think we'd effectively get that by only exporting task-specific functions, as you are already doing below with `addProcessBulkUploadJob`.

I think we could just delete this function and update the import!

```suggestion
```

Reply:
Yes indeed -- that said, `addJob` is not an imported function but rather is invoked against the runner that came out of `run()` (e.g. `runner.addJob()`.

Everything you said is still true, but I think it will mean changing the `jobQueue` to be a class (with state / runner being an instance variable) instead of a series of exported functions.  In turn, I think this would probably be something that should be passed to the express application, so that handlers are passed the queue.

This is probably better design anyway.

Only writing all this to explain why the next set of changes is going to look so wildly different.

Reply:
I believe the alternative is to have a series of exported functions that all rely on a module-scoped `runner` variable... but that feels like it's just a class pretending to not be a class.

Reply:
Update: we talked through this and will stick with the current implementation *but* I'll have it utilize the `db.pool` as the pgpool in order to avoid the overhead of creating a new connection every time a job is added.